File: VendingMachine/Models/Coin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using static VendingMachineApi.Classes.Enumerations;

namespace VendingMachineApi.Models
{
    public class Coin
    {
        public CoinType CoinType { get; set; }
        public int Value { get; set; }

        public int Quantity { get; set; }
    }
}

File: VendingMachineFrontend/src/app/models/coin.ts
export class Coin {
  name: string;
  value: number;
  coinType: number;
  quantity: number;

  constructor(name:string, value:number, coinType:number) {
    this.name = name;
    this.value = value;
    this.coinType = coinType;
    this.quantity = 0;
  }
}
